src/ssh/cert.rs

```diff
@@ -160,25 +160,30 @@ impl Certificate {
     /// println!("{:?}", cert);
     /// ```
     pub fn from_string(s: &str) -> Result<Certificate> {
-        let mut iter = s.split_whitespace();
-
-        let kt_name = iter.next().ok_or(Error::InvalidFormat)?;
-
-        let key_type = KeyType::from_name(kt_name)?;
-        if !key_type.is_cert {
-            return Err(Error::NotCertificate);
-        }
+        let mut iter = s.split_whitespace().skip(1);
 
         let data = iter.next().ok_or(Error::InvalidFormat)?;
-
         let comment = iter.next().map(String::from);
-        let decoded = base64::decode(&data)?;
-        let mut reader = Reader::new(&decoded);
+        let decoded = base64::decode(data)?;
+
+        let mut cert = Certificate::from_bytes(&decoded)?;
+        cert.comment = comment;
+        Ok(cert)
+    }
+
+    /// Reads an SSH certificate from a given byte sequence.
+    ///
+    /// The byte sequence is expected to be the base64 decoded body of the SSH certificate.
+    ///
+    pub fn from_bytes(data: &[u8]) -> Result<Certificate> {
+        let mut reader = Reader::new(&data);
 
         // Validate key types before reading the rest of the data
-        let kt_from_reader = reader.read_string()?;
-        if kt_name != kt_from_reader {
```

Comment:
I'm a little concerned this check appears to have been removed. I need to think more about if this is ok but I seem to remember adding this to catch certificates that do not state the correct type in their preamble